File: dotnet/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example3/Handlers/ManagerHandler.cs
namespace ChainOfResponsibilityDesignPattern.Example3.Handlers;

public class ManagerHandler : BaseSupportHandler
{
    public ManagerHandler() : base("Support Manager") { }

    public override void HandleTicket(SupportTicket ticket)
    {
        // Manager handles all Critical priority tickets
        if (ticket.Priority == TicketPriority.Critical)
        {
            ticket.Resolve(HandlerName, "Escalated to engineering team, provided temporary workaround, and scheduled priority fix.");
        }
        else
        {
            base.HandleTicket(ticket);
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example3/src/Handlers/ManagerHandler.ts
import { SupportTicket, TicketPriority } from "../SupportTicket";
import { BaseSupportHandler } from "../BaseSupportHandler";

export class ManagerHandler extends BaseSupportHandler {
  constructor() { super("Support Manager"); }

  handleTicket(ticket: SupportTicket): void {
    if (ticket.priority === TicketPriority.Critical) {
      ticket.resolve(
        this.handlerName,
        "Escalated to engineering team, provided temporary workaround, and scheduled priority fix."
      );
    } else {
      super.handleTicket(ticket);
    }
  }
}
